README.md

```diff
@@ -5,33 +5,33 @@ Cancel stuck transactions on the Filecoin network.
 ## Usage
 
 ```js
-import { CancelStuckTransactions } from 'cancel-stuck-transactions'
+import { StuckTransactionsCanceller } from 'cancel-stuck-transactions'
 import timers from 'node:timers/promises'
 import fs from 'node:fs/promises'
 
-const cancelStuckTransactions = new CancelStuckTransactions({
+const stuckTransactionsCanceller = new StuckTransactionsCanceller({
   // Pass a storage adapter, so that pending cancellations are persisted across
   // process restarts
-  async store ({ hash, timestamp, from, gasPremium, nonce }) {
-    await fs.writeFile(
-      `transactions/${hash}`,
-      JSON.stringify({ hash, timestamp, from, gasPremium, nonce })
-    )
+  store: {
+    async add ({ hash, timestamp, from, gasPremium, nonce }) {
```

Comment:
Nitpick - feel free to ignore.

Conceptually, `store` is a key-value store using `hash` as the key. I think the following API would make this fact more obvious to users:

```
async add (hash, { timestamp, from, gasPremium, nonce }) {
```

What is the semantics of this method when it comes to repeatedly adding the same key? In other words, what is the expected outcome of the following code:

```js
store.add({hash: 'hash', nonce: 10, /* other data */ })
store.add({hash: 'hash', nonce: 20, /* other data */ })
```

I think `store.set` is a better name. (Other alternatives that come to my mind: `store.store`, `store.save`.)

We could take it one step further and use the same method names as the built-in `Map` object provides:

 - `set(hash, data)`
 - `entries()`
 - `delete(hash)`

Reply:
I didn't think about this too much because it's impossible for two transactions to have the same hash. You're right that `set()` is more correct, but then I think we should rename `addPending(tx)` to `setPending(tx)` as well. Do you think this is worth it now?

> We could take it one step further and use the same method names as the built-in Map object provides:

This could be nice if you wanted to pass `{ store: new Map() }`. I think this would be nice but can be added later on.

Reply:
> I didn't think about this too much because it's impossible for two transactions to have the same hash. You're right that `set()` is more correct, 

I think it's worth improving the `store` API to make it obvious to implementers what the expected semantics are. It's not very clear to me that the `store` API assumes that "it's impossible for two transactions to have the same hash", I see the store API more of a generic key-value store. But maybe that's just me.

> but then I think we should rename `addPending(tx)` to `setPending(tx)` as well. 

I am fine with `addPending`, since at this level of abstraction, we are dealing with full transaction objects where it's obvious that adding the same TX second time is a no-op.

> Do you think this is worth it now?

Yes, it's a quick change that only affects this PR. If we leave the change until later, we would have to rework all consumers of this module, which means more work.
 
Having said that, we are probably spending more time on discussing this change than it's worth it. If you are fine with `store.add`, then let's keep it.

> > We could take it one step further and use the same method names as the built-in Map object provides:
> 
> This could be nice if you wanted to pass `{ store: new Map() }`. I think this would be nice but can be added later on.

The benefit of using a Map-like API is that developers don't have to learn a new shape of a key/value store API, they can use what they are familiar with.

Having said that, this is a minor detail, let's not bikeshed.

Reply:
Ok I'll go with `add()`